CONTRIBUTING.md

```diff
@@ -253,3 +253,28 @@ expanding.
 some housekeeping needs to be done in public API definitions (`.publicApi`
 folder). This can be done automatically via a code fix offered by the public API
 analyzer.
+
+## License requirements
+
+OpenTelemetry .NET is licensed under the [Apache License, Version
+2.0](./LICENSE.TXT).
+
+### Copying files from other projects
+
+OpenTelemetry .NET uses some files from other projects, typically where a binary
+distribution does not exist or would be inconvenient.
+
+The following rules must be followed for PRs that include files from another
+project:
+
+- The license of the file is
+  [permissive](https://en.wikipedia.org/wiki/Permissive_free_software_licence).
+- The license of the file is left in-tact.
+- The contribution is correctly attributed in the [3rd party
```

Comment:
Would you be attributing the existing third-party code in the `THIRD-PARTY-NOTICES.TXT` in some follow-up PR?

Reply:
That's what #5190 was all about. You think any more is needed?

Reply:
I was checking the dotnet runtime repo's `THIRD-PARTY-NOTICES.TXT` and their attributions are more specific where they list out the individual components/file copied along with the project URL that it's copied from.

I was wondering if we want to do that as well instead of having a blanket attribution to "Copyright (c) .NET Foundation and Contributors".

Just a thought. Not a blocker by any means.